onnx/defs/schema.h

```diff
@@ -521,22 +579,24 @@ class OpSchemaRegistry {
    * We wrap it inside a function to avoid the statia initialization order
    * fiasco.
    */
-  static std::unordered_map<std::string, std::map<OperatorSetVersion, OpSchema>>& map();
+  static OpName_Domain_Version_Schema_Map& map();
 
  public:
-  static const std::unordered_map<std::string, std::map<OperatorSetVersion, OpSchema>>& registered_schemas() {
+  static const OpName_Domain_Version_Schema_Map& registered_schemas() {
     return map();
   }
 
-  static const std::unordered_map<std::string, OpSchema> latest_registered_schemas() {
+  static const std::unordered_map<std::string, OpSchema>
+  latest_registered_schemas() {
     // TODO: Do this better with C++11
     std::unordered_map<std::string, OpSchema> latest_map;
     for (auto kv : map()) {
-      latest_map.emplace(kv.first, kv.second.rbegin()->second);
+      for (auto domain_ops : kv.second) {
+        latest_map.emplace(kv.first, domain_ops.second.rbegin()->second);
+      }
```

Comment:
The consequence of getting the domain/name ordering wrong is that the semantics of this function are now very strange: it now *nondeterministically* picks one operator (when there are multiple references of it in multiple domains) and returns it. Not good! Latest registered schemas should instead now return `std::unordered_map<DomainAndName, OpSchema>` or something equivalent; that is an interpretation that actually makes sense.

Reply:
Good point ...

Reply:
this is a great catch! should have <op_name, domain, latest schema> returned.